network/phonebook.go

```diff
@@ -163,14 +175,9 @@ func (mp *MultiPhonebook) GetAddresses(n int) []string {
 	return out
 }
 
-// AddPhonebook adds a Phonebook if it is new
-func (mp *MultiPhonebook) AddPhonebook(p Phonebook) {
+// AddOrUpdatePhonebook adds or updates Phonebook in Pnonebook map
+func (mp *MultiPhonebook) AddOrUpdatePhonebook(name string, p Phonebook) {
```

Comment:
Why wasn't AddPhonebook() enough? It filtered out duplicates and updates to the underlying data were visible because Phonebook is an interface reference type.

Reply:
the underlying map of phonebooks requires key-value pairs.

Reply:
but it doesn't need to be a map. we never need to 'update(phonebookName, phonebookReference)' because it's already a reference and the value is never stale. it already de-duplicated phonebooks added.

Reply:
That is how I started, but it was suggested to manage the DNS bootstrap entries as a map utilizing the MutliPhonebook.